src/error.rs

```diff
@@ -38,8 +53,17 @@ impl Error {
     }
 
     /// Pretty-print this error.
-    pub fn pretty_print(self) {
+    pub fn pretty_print(&self, env: impl Env) {
         use std::io::IsTerminal;
+        let mut err = env.stderr();
+
+        let error = match &self.0.primary {
+            PrimaryError::Clap(e) => {
+                let _ = writeln!(err, "{}", e.render().ansi());
```

Comment:
I need to try this out to see how the errors look, I find it a bit unexpected that there are two separate ways to render errors here and that they may not render the same way.

Reply:
Clap errors are tricky because they do their own styling including their own "error" marker and colors. So we have to print it differently. The context also doesn't make sense. But I wanted to include it in the error type so that we can check for errors in tests (instead of exiting).

Reply:
Ok.